### PR TITLE
make build reproducible

### PR DESCRIPTION
**Pull request by sunny775**
The changes proposed by this PR will make the build process reproducible.
We currently maintain this changes as patches in the [Debian Package](https://salsa.debian.org/js-team/node-d3-time) of d3-time.
It will be super-handy if these changes are integrated in the upstream

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ const config = {
     format: "umd",
     indent: false,
     extend: true,
-    banner: `// ${meta.homepage} v${meta.version} Copyright ${(new Date).getFullYear()} ${meta.author.name}`,
+    banner: `// ${meta.homepage} v${meta.version} Copyright ${(new Date(process.env.SOURCE_DATE_EPOCH ? (process.env.SOURCE_DATE_EPOCH * 1000) : new Date().getTime())).getFullYear()} ${meta.author.name}`,
```
Comment (curran) [inline rollup.config.js:13]: Could be a bit simpler:
```suggestion
    banner: `// ${meta.homepage} v${meta.version} Copyright ${(process.env.SOURCE_DATE_EPOCH ? new Date(process.env.SOURCE_DATE_EPOCH * 1000) : new Date).getFullYear()} ${meta.author.name}`,
```

Comment (Fil) [reply rollup.config.js:13]:
```suggestion
    banner: `// ${meta.homepage} v${meta.version} Copyright ${(new Date(process.env.SOURCE_DATE_EPOCH * 1000 || Date.now()).getFullYear()} ${meta.author.name}`,
```

:p

**Review by Fil — APPROVED** [review]
Should this be done on all d3 modules or only on d3-time?

**Comment by mbostock** [comment]
This should be pulling the copyright year from the LICENSE file, not making it up at runtime.

**Comment by mbostock** [comment]
I plan on doing something like this:

```js
// Extract copyrights from the LICENSE.
const copyrights = fs.readFileSync("./LICENSE", "utf-8")
  .split(/\n/g)
  .filter(line => /^copyright\s+/i.test(line))
  .map(line => line.replace(/^copyright\s+/i, ""));
```

And then

```js
banner: `// ${meta.name} v${meta.version} Copyright ${copyrights.join(", ")}`
```

**Comment by sunny775** [comment]
@mbostock , pulling the copyright year from the LICENSE file seems a better way to make the build reproducible. Thanks a lot

**Comment by sunny775** [comment]
@Fil It will be very nice to have 100% reproducible builds, like doing this on all d3 modules.

**Pull request closed without merging**